src/free_functions.rs

```diff
@@ -88,16 +89,13 @@ pub fn aview1<A>(xs: &[A]) -> ArrayView1<A> {
 }
 
 /// Create a two-dimensional array view with elements borrowing `xs`.
+///
+/// **Panics** if the product of non-zero axis lengths overflows `isize`.
 pub fn aview2<A, V: FixedInitializer<Elem=A>>(xs: &[V]) -> ArrayView2<A> {
     let cols = V::len();
     let rows = xs.len();
-    let data = unsafe {
-        slice::from_raw_parts(xs.as_ptr() as *const A, cols * rows)
-    };
-    let dim = Ix2(rows, cols);
-    unsafe {
-        ArrayView::from_shape_ptr(dim, data.as_ptr())
-    }
+    let data = unsafe { slice::from_raw_parts(xs.as_ptr() as *const A, cols * rows) };
+    ArrayView::from_shape(Ix2(rows, cols), data).unwrap()
 }
```

Comment:
I think this function should only have a check if `size_of::<V>() == 0` and no possible panic otherwise. In the case when `size_of::<V>() > 0`, we have an in-memory slice which we know Rust supports and can index into, and thus we can too. I think we can infer that our conditions are met.

Reply:
Same for aview2_mut